src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/TensorsInfo.cs

```diff
@@ -469,19 +474,46 @@ protected virtual void Dispose(bool disposing)
             _disposed = true;
         }
 
+        private static int[] ConvertDimension(int[] dimension)
+        {
+            if (dimension == null) {
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "dimension is null");
+            }
+
+            if (dimension.Length > Tensor.RankLimit) {
+                throw new IndexOutOfRangeException("Max rank limit is " + Tensor.RankLimit);
+            }
+
+            int[] dim = new int[Tensor.RankLimit];
+            int i;
+
+            for (i = 0 ; i < dimension.Length ; i++) {
+                dim[i] = dimension[i];
+            }
+            for (; i < Tensor.RankLimit ; i++) {
+                dim[i] = 0;
+            }
+
+            return dim;
+        }
+
         private void UpdateInfoHandle(IntPtr handle, int idx, string name, TensorType type, int[] dimension)
         {
+            int[] dim;
+
             if (handle != IntPtr.Zero)
             {
                 NNStreamerError ret = NNStreamerError.None;
 
+                dim = ConvertDimension(dimension);
```

Comment:
The length of dimension passed to UpdateInfoHandle() is always 16.

Reply:
Updated it :)
Thank you 👍